### PR TITLE
Fix alias_method_chain

### PR DESCRIPTION
**Pull request by keymastervn**
## Change
 - Fix alias_method_chain
 - Bump version to 3.2.23

### CODE REVIEW

**Review by phamtuan2993 — APPROVED** [review]
noice

File: Rakefile [hunk 1]
```diff
@@ -10,5 +10,6 @@ Rake::TestTask.new(:test) do |t|
   t.libs << '.'
   t.pattern = 'test/**/*_test.rb'
   t.verbose = true
+  t.warning = false
```
Comment (chipset508) [inline Rakefile:13]: can you explain this change plz

Comment (keymastervn) [reply Rakefile:13]: [Before](https://app.circleci.com/pipelines/github/Thinkei/xero_gateway/15/workflows/27497641-24c6-40df-83c1-eb44f7659f4c/jobs/224/parallel-runs/0/steps/0-105) and [after](https://app.circleci.com/pipelines/github/Thinkei/xero_gateway/40/workflows/70c68c42-5c26-44b7-b82a-aeb420587f07/jobs/256/parallel-runs/0/steps/0-106) anh 

Purging these unresolved warning in `rake test`
```
home/circleci/xero_gateway/lib/xero_gateway/invoice.rb:226: warning: duplicated when clause is ignored
/home/circleci/xero_gateway/lib/xero_gateway/invoice.rb:227: warning: duplicated when clause is ignored
/home/circleci/xero_gateway/lib/xero_gateway/invoice.rb:108: warning: method redefined; discarding old contact
/home/circleci/xero_gateway/lib/xero_gateway/invoice.rb:128: warning: method redefined; discarding old line_items
/home/circleci/xero_gateway/lib/xero_gateway/bank_transaction.rb:102: warning: method redefined; discarding old line_items
/home/circleci/xero_gateway/lib/xero_gateway/credit_note.rb:209: warning: duplicated when clause is ignored
/home/circleci/xero_gateway/lib/xero_gateway/credit_note.rb:210: warning: duplicated when clause is ignored
/home/circleci/xero_gateway/lib/xero_gateway/credit_note.rb:103: warning: method redefined; discarding old contact
/home/circleci/xero_gateway/lib/xero_gateway/credit_note.rb:123: warning: method redefined; discarding old line_items
/home/circleci/xero_gateway/lib/xero_gateway/credit_note.rb:171: warning: method redefined; discarding old save
/home/circleci/xero_gateway/lib/xero_gateway/credit_note.rb:159: warning: previous definition of save was here
/home/circleci/xero_gateway/lib/xero_gateway/manual_journal.rb:99: warning: method redefined; discarding old journal_lines
/home/circleci/xero_gateway/lib/xero_gateway/organisation.rb:24: warning: `*' interpreted as argument prefix
/home/circleci/xero_gateway/lib/xero_gateway/tax_rate.rb:18: warning: `*' interpreted as argument prefix
/home/circleci/xero_gateway/lib/xero_gateway/currency.rb:11: warning: `*' interpreted as argument prefix
/home/circleci/xero_gateway/lib/xero_gateway/gateway.rb:16: warning: method redefined; discarding old logger
...
../home/circleci/xero_gateway/lib/xero_gateway/gateway.rb:620:in `get_bank_transaction': warning: URI.escape is obsolete
./home/circleci/xero_gateway/lib/xero_gateway/gateway.rb:823:in `get_contact': warning: URI.escape is obsolete
.../home/circleci/xero_gateway/lib/xero_gateway/gateway.rb:475:in `get_credit_note': warning: URI.escape is obsolete
/home/circleci/xero_gateway/lib/xero_gateway/gateway.rb:475:in `get_credit_note': warning: URI.escape is obsolete
```

Comment (chipset508) [reply Rakefile:13]: those are warning only right or it break something? i not sure about the reason why we need to clear warning log

Comment (keymastervn) [reply Rakefile:13]: They are just warnings, hmm, let me re-enable the warning. Feel like we are logging them but no follow-up action defined is a pitfall

**Review by chipset508 — APPROVED** [review]

**Review by phamtuan2993 — APPROVED** [review]

File: lib/oauth/oauth_consumer.rb [hunk 2]
```diff
@@ -7,7 +7,7 @@ def http_with_ssl_client_certificates(*args)
       end
     end
 
-    alias_method :ssl_client_certificates, :http
-    alias_method :http, :ssl_client_certificates
+    alias_method :http_without_ssl_client_certificates, :http
```
Comment (vinh-tony) [inline lib/oauth/oauth_consumer.rb:10]: hmm, I'm about confused about what those LOC going to do. 
- Remove alias_method for `ssl_client_certificates` => Have you checked that we will be safe after remove this alias_method? (I mean this method `ssl_client_certificates` going to be undefined after merging, is it safe?
- I guess we're gonna force to ssl for all http/https. Can we refactor those LOC to:
```ruby
alias_method :http_without_ssl_client_certificates, :http_with_ssl_client_certificates
alias_method :http, :http_with_ssl_client_certificates
```

Comment (phamtuan2993) [reply lib/oauth/oauth_consumer.rb:10]: haha, this is actually a fix, not a change a, previously it was
alias_method_chain :http, :ssl_client_certificates
ref [alias_method_chain](https://apidock.com/rails/ActiveSupport/CoreExtensions/Module/alias_method_chain)

but I was misunderstood and make it wrong, and now a @keymastervn is fixing it a :D

Comment (keymastervn) [reply lib/oauth/oauth_consumer.rb:10]: @vinh-tony 
The method `ssl_client_certificates` has never been implemented anh, let's put an eye to [this](https://github.com/oauth-xx/oauth-ruby/blob/v0.4.0/lib/oauth/consumer.rb#L91) `http` method. The code is creating `http_with/without_ssl_client_certificates` 

This code is a monkey patch, just weird, is creating 2 versions of `http` for supporting ssl in its first intent and backward compatible to `http` (a.k.a `http_without_ssl_client_certificates`)

**Review by vinh-tony — APPROVED** [review]

**Pull request merged**